content/2022-04-29-firmware-updates-part-1/index.md

```diff
@@ -0,0 +1,160 @@
++++
+title = "Firmware updates, part 1: Bootloader"
+extra.author = "lulf"
++++
+
+This is the first post in a series about doing device firmware updates (DFU) over the air (OTA) and continuous delivery of firmware for embedded devices. We'll explore the different parts of a complete end to end system with this capability.
+
+This post will be about a fundamental component in such a system, the bootloader.
+
+<!-- more -->
+
+# Background
+
+For a connected devices to be maintainable at scale, they must be able to update themselves just like any other software. However, managing a fleet of occasionally connected devices with little bandwidth require a different approach to delivering software updates to regular applications. One goal for Drogue IoT is to support the entire software update workflow on tiny devices, all the way from building your firmware to delivering it to the device, just like any other modern application.
+
+# Bootloader
+
+A fundamental component in an updatable system is the ability to boot different versions of an application. Making a generic bootloader is hard, because there is a big number of different possible device configurations. The complexity and size of a bootloader is also determined by it's functionality: a bootloader being able to retrieve firmware from a network has a bigger footprint than one that only loads an application from a fixed location in flash.
```

Comment:
```suggestion
A fundamental component in an updatable system is the ability to boot different versions of an application. Making a generic bootloader is hard, because there is a large number of different possible device configurations. The complexity and size of a bootloader is also determined by its functionality: a bootloader being able to retrieve firmware from a network has a bigger footprint than one that only loads an application from a fixed location in flash.
```